discovery/src/main/java/org/ethereum/beacon/discovery/enr/NodeRecord.java

```diff
@@ -0,0 +1,192 @@
+package org.ethereum.beacon.discovery.enr;
+
+import com.google.common.base.Objects;
+import org.javatuples.Pair;
+import org.web3j.rlp.RlpEncoder;
+import org.web3j.rlp.RlpList;
+import org.web3j.rlp.RlpString;
+import org.web3j.rlp.RlpType;
+import tech.pegasys.artemis.util.bytes.Bytes32;
+import tech.pegasys.artemis.util.bytes.BytesValue;
+import tech.pegasys.artemis.util.uint.UInt64;
+
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Ethereum Node Record
+ *
+ * <p>Node record as described in <a href="https://eips.ethereum.org/EIPS/eip-778">EIP-778</a>
+ */
+public class NodeRecord {
+  // Compressed secp256k1 public key, 33 bytes
+  public static String FIELD_PKEY_SECP256K1 = "secp256k1";
+  // IPv4 address
+  public static String FIELD_IP_V4 = "ip";
+  // TCP port, integer
+  public static String FIELD_TCP_V4 = "tcp";
+  // UDP port, integer
+  public static String FIELD_UDP_V4 = "udp";
+  // IPv6 address
+  public static String FIELD_IP_V6 = "ip6";
+  // IPv6-specific TCP port
+  public static String FIELD_TCP_V6 = "tcp6";
+  // IPv6-specific UDP port
+  public static String FIELD_UDP_V6 = "udp6";
+
+  private UInt64 seq;
+  // Signature
+  private BytesValue signature;
+  // optional fields
+  private Map<String, Object> fields = new HashMap<>();
+
+  private EnrSchemeInterpreter enrSchemeInterpreter;
```

Comment:
Creating a scheme doesn't have big resource impact; moreover, it happens infrequent. Let's maybe do this on demand and get rid of this field?

Reply:
I already use this for tests to have two interpreters: with verification and without. I see no better approach while having feature of verification override.